Makefile

```diff
@@ -44,6 +46,11 @@ BUILDFLAGS :=  -ldflags \
 # Some tests expect default values for version.*, so just use the config package values there.
 TEST_BUILDFLAGS :=  -ldflags "$(BUILD_TIME_CONFIG_FLAGS)"
 
+ifneq ($(GO_VERSION),$(CATALOG_GO_VERSION))
+$(error The local version of GO doesn't match the one in pipeline catalog. Please set Go to $(CATALOG_GO_VERSION) )
```

Comment:
Can we only warn, and not throw error?

Reply:
I thought error would be better to "enforce" the user to upgrade. Why you think warn is better?

Reply:
We dont have updatebot in place yet (for triggering builds), so just failing it for now, will break things in unexpected way, warn is ok for now ...
Also updatebot may not work the way I expect it to - it may still require some manual touch ...

Reply:
@ankitm123 Great, done